contracts/contracts/ChatOracle.sol

```diff
@@ -228,14 +228,17 @@ contract ChatOracle {
         string memory errorMessage
     ) public onlyWhitelisted {
         require(!isPromptProcessed[promptId], "Prompt already processed");
-        isPromptProcessed[promptId] = true;
         IChatGpt(callbackAddresses[promptId]).onOracleLlmResponse(
             promptCallBackId,
             response,
             errorMessage
         );
     }
 
+    function markPromptAsProcessed(uint promptId) public onlyWhitelisted {
+        isPromptProcessed[promptId] = true;
```

Comment:
In the previous function `addResponse` the `IChatGpt(callbackAddresses[promptId]).onOracleLlmResponse()` can fail and the oracle retries this on restart. This is a suggestion from @kgrofelnik  to have a separate tx to mark it as done